buildkitd/buildkitd.go

```diff
@@ -167,17 +167,41 @@ func MaybeRestart(ctx context.Context, console conslogging.ConsoleLogger, image
 	return nil
 }
 
+// RemoveExited removes any stopped or exited buildkitd containers
+func RemoveExited(ctx context.Context) error {
+	started, err := IsStarted(ctx)
```

Comment:
I don't think we need to check here too because we already check in the caller of `Start`.

Reply:
I agree with that; I added in this check in case for some reasons `RemoveExited` was called in a different code path. If the container is up and running RemoveExited will return an error saying the container is running and needs to be stopped (or we use the `--force` flag).

This is probably fine since it'll return an error if it's used in a different code path, and we could change the behaviour then.